lib/ctblfuns.gi

```diff
@@ -1666,7 +1666,9 @@ InstallMethod( ScalarProduct,
     x2:= ValuesOfClassFunction( x2 );
     scpr:= 0;
     for i in [ 1 .. Length( x1 ) ] do
-      scpr:= scpr + x1[i] * GaloisCyc( x2[i], -1 ) * weight[i];
+      if x1[i]<>0 and x2[i]<>0 then
+	scpr:= scpr + x1[i] * GaloisCyc( x2[i], -1 ) * weight[i];
```

Comment:
looks weird in the webinterface -- uses tabs for indentation?

Reply:
Yes -- my editor (vim) seems to introduce tabs if the file already contains tabs. I have replaced my tabs with spaces.

Reply:
Unfortunately I still see tabs in this line (the other two seem to be fixed, though)